staging/src/k8s.io/api/networking/v1/types.go

```diff
@@ -344,22 +344,59 @@ type HTTPIngressRuleValue struct {
+	// list of labels in the path split by the '/' separator. A request is a
+	// match for path p if every p is an element-wise prefix of p of the
+	// request path. Note that if the last element of the path is a substring
+	// of the last element in request path, it is not a match (e.g. /foo/bar
+	// matches /foo/bar/baz, but does not match /foo/barbaz).
+	PathTypePrefix = PathType("Prefix")
+
+	// PathTypeImplementationSpecific matching is up to the IngressClass.
+	// Implementations are not required to support ImplementationSpecific
+	// matching.
+	PathTypeImplementationSpecific = PathType("ImplementationSpecifc")
+)
+
+// HTTPIngressPath associates a path regex with a backend. Incoming urls
+// matching the path are forwarded to the backend.
 type HTTPIngressPath struct {
-	// Path is an extended POSIX regex as defined by IEEE Std 1003.1,
-	// (i.e this follows the egrep/unix syntax, not the perl syntax)
-	// matched against the path of an incoming request. Currently it can
-	// contain characters disallowed from the conventional "path"
-	// part of a URL as defined by RFC 3986. Paths must begin with
-	// a '/'. If unspecified, the path defaults to a catch all sending
-	// traffic to the backend.
+	// Path is an extended POSIX regex as defined by IEEE Std 1003.1, (i.e this
+	// follows the egrep/unix syntax, not the perl syntax) matched against the
+	// path of an incoming request. Currently it can contain characters
+	// disallowed from the conventional "path" part of a URL as defined by RFC
+	// 3986. Paths must begin with a '/'. If unspecified, the path defaults to a
+	// catch all sending traffic to the backend.
+	// +optional
+	Path string `protobuf:"bytes,1,opt,name=path"`
+
+	// PathType determines the interpretation of the Path matching. PathType can
+	// be one of the following values:
+	// * Exact: Matches the URL path exactly.
+	// * Prefix: Matches based on a URL path prefix split by '/'. Matching is
+	//   done on a path element by element basis. A path element refers is the
+	//   list of labels in the path split by the '/' separator. A request is a
+	//   match for path p if every p is an element-wise prefix of p of the
+	//   request path. Note that if the last element of the path is a substring
+	//   of the last element in request path, it is not a match (e.g. /foo/bar
+	//   matches /foo/bar/baz, but does not match /foo/barbaz).
+	// * ImplementationSpecific: Interpretation of the Path matching is up to
+	//   the IngressClass. Implementations are not required to support
+	//   ImplementationSpecific matching.
 	// +optional
-	Path string `json:"path,omitempty" protobuf:"bytes,1,opt,name=path"`
+	PathType PathType `protobuf:"bytes,2,opt,name=pathType,casttype=PathType"`
 
 	// Backend defines the referenced service endpoint to which the traffic
 	// will be forwarded to.
-	Backend IngressBackend `json:"backend" protobuf:"bytes,2,opt,name=backend"`
+	Backend IngressBackend `protobuf:"bytes,3,opt,name=backend"`
```

Comment:
I can't remember, is it backwards compatible to change this from 2-3? I wonder if this should instead just remain 2 and we make the newer item 3.

Reply:
Good catch, I'm not sure, but switched it back to be safe.